orion-commons/src/main/java/com/pinterest/orion/utils/TeletraanClient.java

```diff
@@ -0,0 +1,280 @@
+        }
+    }
+
+    /**
+     * Parse the response entity to check if the host is terminated.
+     * @param responseEntity HttpEntity from the response
+     * @return boolean indicating if the host is terminated
+     */
+    protected boolean ParseHostTerminatedStatus(HttpEntity responseEntity) {
+        try {
+            String responseEntityString = EntityUtils.toString(responseEntity);
+            JsonParser parser = new JsonParser();
+            JsonArray responseEntityArray = (JsonArray) parser.parse(responseEntityString);
+            if (responseEntityArray.size() == 0) {
+                // No record found for this host. Host has been terminated.
+                return true;
+            }
+            return false;
+        } catch (Exception e) {
+            logger.severe("Error in parsing host status: " + e);
+            return false;
+        }
+    }
+
+    /**
+     * Check the status of a host using the Teletraan API.
+     * @param httpClient
+     * @param hostName
+     * @param overrideTeletraanToken
+     * @return HttpEntity from the response.
+     * The entity is a json array of host status objects. The first object is the most recent status.
+     * If the array is empty, the host is terminated or not exist.
+     */
+    public HttpEntity checkHostStatus(
+            CloseableHttpClient httpClient,
+            String hostName,
+            String overrideTeletraanToken) {
+        try {
+            String checkHostStatusUrl = getCheckHostStatusUrl(hostName);
+            HttpGet hostStatusCheckRequest = new HttpGet(checkHostStatusUrl);
+            hostStatusCheckRequest.setHeader("Content-Type", "application/json");
+            hostStatusCheckRequest.setHeader("Authorization", getTokenHeader(overrideTeletraanToken));
+            logger.info(String.format("Checking host status for %s via Teletraan API: %s",
+                    hostName, checkHostStatusUrl));
+            CloseableHttpResponse response = httpClient.execute(hostStatusCheckRequest);
+            if (response.getStatusLine().getStatusCode() != HttpURLConnection.HTTP_OK) {
+                logger.severe(String.format(
+                        "Failed to get host status for %s via Teletraan API: %s. Status: %s",
+                        hostName, checkHostStatusUrl, response.getStatusLine().getStatusCode()));
+                return null;
+            }
+            return response.getEntity();
+        } catch (Exception e) {
+            logger.severe(String.format(
+                    "Error in host status check for %s via Teletraan API. Exception: %s", hostName, e
+            ));
+            return null;
+        }
+    }
+
+    /**
+     * Check if the instance is pending termination using the Teletraan API.
+     * @param httpClient
+     * @param hostName
+     * @param overrideTeletraanToken
+     * @return boolean indicating if the instance is pending termination
+     */
+    public boolean isInstancePendingTermination(
```

Comment:
Why not have one function for this status check? There was another one above.

Reply:
Which one? isHostPendingTermination is getting the status and parse the response to know the host status. isHostTerminated is checking the host cannot be found. They cannot be merged together. The shared part (HTTP call) is in getHostStatusHistory

Reply:
The reason I asked is both functions return a boolean.

Reply:
One is for parsing http entity to get the result, another one is for to make http call itself. Any suggestion for method names? I use isInstancePendingTermination/IsInstancePendingTermination as you suggested.